File: backend/src/main/java/com/familytree/app/relationship/Relationship.java
package com.familytree.app.relationship;

import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDate;

@Entity
@Table(name = "relationships")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Relationship {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    // References person ids (kept loosely coupled — no JPA join to Person here)
    @Column(nullable = false)
    private Long personAId;

    @Column(nullable = false)
    private Long personBId;

    @Enumerated(EnumType.STRING)
    private RelationType type;

    private LocalDate startDate;

    private LocalDate endDate;

    public enum RelationType { MARRIED, PARTNER, DIVORCED, ENGAGED }
}

File: mobile/lib/models/relationship.dart
class Relationship {
  final int? id;
  final int personAId;
  final int personBId;
  final String? type; // MARRIED, PARTNER, DIVORCED, ENGAGED
  final String? startDate;
  final String? endDate;

  Relationship({
    this.id,
    required this.personAId,
    required this.personBId,
    this.type,
    this.startDate,
    this.endDate,
  });

  factory Relationship.fromJson(Map<String, dynamic> json) {
    return Relationship(
      id: json['id'],
      personAId: json['personAId'],
      personBId: json['personBId'],
      type: json['type'],
      startDate: json['startDate'],
      endDate: json['endDate'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'personAId': personAId,
      'personBId': personBId,
      'type': type,
      'startDate': startDate,
      'endDate': endDate,
    };
  }
}
